services/packagist/packagist-base.js

```diff
@@ -5,9 +5,12 @@ import { isStable, latest } from '../php-version.js'
 const packageSchema = Joi.array().items(
   Joi.object({
     version: Joi.string().required(),
-    require: Joi.object({
-      php: Joi.string(),
-    }),
+    require: Joi.alternatives(
+      Joi.object({
+        php: Joi.string(),
+      }).required(),
+      Joi.string().valid('__unset')
```

Comment:
I'm not terribly familiar with the PHP ecosystem, but are there any other well known values we could potentially see here or is `__unset` essentially it based on what we know today?

Reply:
I gather adding a live test would probably be overkill, but would it be worth leaving an inline comment with a link to the issue?

Reply:
Right - having looked at it, we are running the validation on the uncompressed response which is why the `__unset` values are still in there at the point we validate. I misremembered that we had done something like I suggested in https://github.com/badges/shields/pull/6508#issuecomment-855295658 but we never did.
I think I'd like to revisit this code as I don't think what we are doing now is the best solution but I don't think I want to shave this specific yak tonight :cow2: . I'm going to merge this and punt that to an issue.

Reply:
I've written it up in #7783